source/daplink/interface/swd_host.c

```diff
@@ -836,13 +836,16 @@ uint8_t swd_uninit_debug(void)
     // Clear (CSYSPWRUPREQ | CDBGPWRUPREQ) in DP_CTRL_STAT register
     // First read the register in order to clear the specific bits
     uint32_t ctrl_stat_read;
-    swd_read_dp(DP_CTRL_STAT, &ctrl_stat_read);
-    ctrl_stat_read &= ~(CSYSPWRUPREQ | CDBGPWRUPREQ);
-    
-    if (!swd_write_dp(DP_CTRL_STAT, ctrl_stat_read)) {
+    if (!swd_write_dp(DP_CTRL_STAT, 0x0)) {
         return 0;
     }
     
+    do {
```

Comment:
So you've added a busy wait to check that the debugger is actually uninitialized before exiting?
This should probably be in a commit of its own, or in the other commit.